analyses/cell-type-wilms-tumor-06/notebook_template/06_cnv_infercnv_exploration.Rmd

```diff
@@ -0,0 +1,765 @@
+
+```{r fig.width=5, fig.height=5, out.width='40%', out.height='40%', warning=FALSE}
+for(i in 1:22){
+  print(visualize_density(srat_i6, features=glue::glue("proportion_cnv_chr", i), group.by = "fetal_kidney_predicted.compartment"))
+}
+```
+
+
+#### DotPlot
+
+The `Dotplot` representation summarizes the percentage of cells in each compartment with cnv in each of the 22 chromosomes. 
+
+The HMM-i3 prediction model with intra-patient reference (composed of immune and endothelium cells) perform really well in identifying Wilms tumor relevant CNV.
+No false positive CNV are observed in the immune and endothelium compartment. 
+
+```{r fig.width=15, fig.height=5, out.width='100%'}
+
+DotPlot(srat_i6, features =  colnames(srat_i6@meta.data)[grepl("has_cnv_chr", colnames(srat_i6@meta.data))], group.by = "fetal_kidney_predicted.compartment", assay = "SCT") + 
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
+
+```
+
+#### CNV score
+
+Finally, we try to calculate  single CNV score and assess its potential in iditenfying cells with CNV versus normal cells without CNV.
+
+We simply checked for each chromosome if the cell `has_cnv_chr`. 
+Would the cell have one or more chromosome with CNV, the global `has_cnv_score` will be TRUE.
+Else, the cell will have a `has_cnv_score` set to FALSE.
+
+
+```{r fig.width=20, fig.height=5, out.width='100%', out.height='100%', warning=FALSE}
+meta <- srat_i6@meta.data
+
+meta$has_cnv_score <- rowSums(meta[,grepl("has_cnv_chr", colnames(meta))])
+meta$has_cnv_score[meta$has_cnv_score > 1] <- "CNV"
+
+# we know that some immune cells might have false positive CNV 
+meta$has_cnv_score[meta$has_cnv_score %in% c(0,1)] <- "no CNV"
+
+srat_i6 <- AddMetaData(srat_i6, meta$has_cnv_score, col.name =  "has_cnv_score")
+
+```
+
+
+```{r fig.width=20, fig.height=8, out.width='100%'}
+
+visualize_metadata(srat_i6, "has_cnv_score", "seurat_clusters")
+
+```
+
+
+```{r fig.width=20, fig.height=8, out.width='100%'}
+
+visualize_metadata(srat_i6, "has_cnv_score", "fetal_kidney_predicted.compartment")
+
+
+```
+
+
+
+
+### HMM-i3 CNV prediction model with an inter-patient reference of immune cells (endothelial and immune)
+
+We lastly explore `infercnv` results generated with a pull of immune and endothelial cells from different Wilms tumor samples as reference, using a HMM-i3 prediction model.
+
+The selection of normal cells to be spiked-in as a normal reference prior running `infercnv` has been performed in `06a_build-normal-cell_reference.R.
+We selected samples that haven't been pre-treated with chemotherapies. Indeed, even if normal cells shouldn't be affected by chemotherapy in terms of CNV, we are not sure how chemotherapies can affect the CNV profile of normal cells.
+We thus decided to only take endothelial and immune cells from non-treated samples to build the reference of normal cells.
+
+
+#### Load the `Seurat` object
```

Comment:
@sjspielman I think you are right, might e an `Error` in the construction if the `srat` object but I can't figure it out... I am not able to reproduce the `Error` 🤔 

What about removing the exploration of the `infercnv with `_pull_ = _pool_ of inter-patient normal cells as reference for now, as it can be quite controversial on its own and I am able to run `infercnv` on 39/40 samples with a intra-sample normal reference? 

Might be something that we investigate then in a later PR?

Reply:
I tried to adapt the `Seurat` object construction to make it safer and hopefully avoir `Error`. Let me know if it helps, it is running from my side 🤞

Reply:
Noting we found the issue - we were using input 02b files generated from different data releases.